pilot/pkg/model/validation.go

```diff
@@ -1532,11 +1524,11 @@ func validateRouteDestinations(weights []*networking.RouteDestination) (errs err
 		}
 		errs = appendErrors(errs, validateDestination(weight.Destination))
 		errs = appendErrors(errs, ValidatePercent(weight.Weight))
+		if len(weights) > 1 && weight.Weight < 1 {
+			errs = multierror.Append(errs, errors.New("positive weight is required"))
+		}
 		totalWeight += weight.Weight
 	}
-	if len(weights) > 1 && totalWeight != 100 {
```

Comment:
why are we removing this? we still need total weights to equal 100, right?

Reply:
This isn't required for TCP clusters in Envoy.

https://github.com/envoyproxy/envoy/blob/10625c596325c2bdfea47e3063161aa5edec495f/api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto#L159-L161

Reply:
Will do. This is true only for TCP clusters and not HTTP though.

Reply:
I've created istio/api#647 to address this.